setup.py

```diff
@@ -134,7 +134,8 @@ def run(self):
             'six',
             'pytest-xdist',
             'virtualenv',
-        ]
+        ],
+        'toml': ['toml'],
```

Comment:
shouldn't this be
```suggestion
        'toml': ['coverage[toml] >= WHEN_COVERAGE_GOT_TOML_SUPPORT'],
```

Reply:
this way if coverage changes their toml parser, it will still work

Reply:
I just got the same idea, and tested it locally. But it does not work for me.

This is what I get:
```
$ pip install .[toml]
Processing /home/krys/Python_Projects/.contributions/pytest-cov
Collecting coverage>=4.4
  Using cached coverage-5.1-cp36-cp36m-manylinux1_x86_64.whl (227 kB)
Collecting pytest>=3.6
  Using cached pytest-5.4.3-py3-none-any.whl (248 kB)
Collecting py>=1.5.0
  Using cached py-1.8.1-py2.py3-none-any.whl (83 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.3-py2.py3-none-any.whl (30 kB)
Collecting more-itertools>=4.0.0
  Using cached more_itertools-8.3.0-py3-none-any.whl (44 kB)
Collecting attrs>=17.4.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting pluggy<1.0,>=0.12
  Using cached pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting packaging
  Using cached packaging-20.4-py2.py3-none-any.whl (37 kB)
Collecting importlib-metadata>=0.12; python_version < "3.8"
  Using cached importlib_metadata-1.6.0-py2.py3-none-any.whl (30 kB)
Collecting pyparsing>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Using legacy setup.py install for pytest-cov, since package 'wheel' is not installed.
Installing collected packages: coverage, py, wcwidth, more-itertools, attrs, zipp, importlib-metadata, pluggy, pyparsing, six, packaging, pytest, pytest-cov
    Running setup.py install for pytest-cov ... done
Successfully installed attrs-19.3.0 coverage-5.1 importlib-metadata-1.6.0 more-itertools-8.3.0 packaging-20.4 pluggy-0.13.1 py-1.8.1 pyparsing-2.4.7 pytest-5.4.3 pytest-cov-2.9.0 six-1.15.0 wcwidth-0.2.3 zipp-3.1.0
```

No `toml` .. but when I comment out `'coverage>=4.4'` in the `install_requires` section it does as we want: 

```
$ pip install .[toml]
Processing /home/krys/Python_Projects/.contributions/pytest-cov
Collecting pytest>=3.6
  Using cached pytest-5.4.3-py3-none-any.whl (248 kB)
Collecting coverage[toml]>=5
  Using cached coverage-5.1-cp36-cp36m-manylinux1_x86_64.whl (227 kB)
Collecting packaging
  Using cached packaging-20.4-py2.py3-none-any.whl (37 kB)
Collecting pluggy<1.0,>=0.12
  Using cached pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting attrs>=17.4.0
  Using cached attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting more-itertools>=4.0.0
  Using cached more_itertools-8.3.0-py3-none-any.whl (44 kB)
Collecting py>=1.5.0
  Using cached py-1.8.1-py2.py3-none-any.whl (83 kB)
Collecting importlib-metadata>=0.12; python_version < "3.8"
  Using cached importlib_metadata-1.6.0-py2.py3-none-any.whl (30 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.3-py2.py3-none-any.whl (30 kB)
Collecting toml; extra == "toml"
  Using cached toml-0.10.1-py2.py3-none-any.whl (19 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting pyparsing>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting zipp>=0.5
  Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
Using legacy setup.py install for pytest-cov, since package 'wheel' is not installed.
Installing collected packages: six, pyparsing, packaging, zipp, importlib-metadata, pluggy, attrs, more-itertools, py, wcwidth, pytest, toml, coverage, pytest-cov
    Running setup.py install for pytest-cov ... done
Successfully installed attrs-19.3.0 coverage-5.1 importlib-metadata-1.6.0 more-itertools-8.3.0 packaging-20.4 pluggy-0.13.1 py-1.8.1 pyparsing-2.4.7 pytest-5.4.3 pytest-cov-2.9.0 six-1.15.0 toml-0.10.1 wcwidth-0.2.3 zipp-3.1.0
```

I think `coverage[toml]` is completely skipped because `coverage` itself is a install dependency.

Reply:
I tried to achieve  this with `poetry` too .. but it does not work also.

Reply:
@graingert After some thinking, googling and testing I think it is impossible to achieve.

Reply:
can you post the links you found?

Reply:
@Cielquan https://github.com/pypa/pip/issues/4957

Reply:
I found issues on stackoverflow .. dont have the links anymore.

But the pip issue you found is good.

Reply:
As mentioned by @hukkinj1 the above mentioned pip issue is closed. I tested it again with a fresh venv and pip 20.3 and pip 21.0.1. Both worked just fine.